src/backends/caffe2/netdef_backend.cc

```diff
@@ -137,6 +137,10 @@ NetDefBackend::CreateExecutionContexts(
         Run(runner_idx, payloads, func);
       }));
 
+  if (Config().has_model_warm_up()) {
```

Comment:
I think we will need to have warmup as one of the "phases" for SetConfiguredScheduler because we need to init to happen before warmup and also need the warmup runs to be trigger by the scheduler thread (as normal runs are done).

Reply:
Do you mean just moving this part into `SetConfiguredScheduler()`? Or actually sending the sample requests to backend context via `Scheduler::Enqueue()`? I wasn't doing the latter because we don't have strong guarantee that all contexts will be run if the request is dispatched by scheduler.